astroid/raw_building.py

```diff
@@ -177,6 +177,7 @@ def register_arguments(func: nodes.FunctionDef, args: list | None = None) -> Non
             func.set_local(func.args.vararg, func.args)
         if func.args.kwarg:
             func.set_local(func.args.kwarg, func.args)
+    assert args is not None
```

Comment:
This is a little stressful :D I'd rather have a type ignore

Reply:
How about adding this instead?

```python
    if args is None:
        return
```

Reply:
Or even better:

```python
    if args is None:
        if func.args.vararg:
            func.set_local(func.args.vararg, func.args)
        if func.args.kwarg:
            func.set_local(func.args.kwarg, func.args)
        args = func.args.args
        if args is None:
            return
```